main.nf

```diff
@@ -0,0 +1,932 @@
+        -o ${name}_realign.bam \\
+        -l INFO
+    """
+}
+
+/*
+ * Step 7 - Determine quality metrics of mapped BAM files using QualiMap 2
+ * 
+*/ 
+process qualiMap {
+    tag "${name}"
+    publishDir "${params.outdir}/qualimap/qualimap", mode: 'copy'
+    
+    input:
+    set val(name), file(realign_bam), file(realign_bam_ind) from bam_metrics
+
+    output:
+    file "${name}_qualimap" into qualimap_results
+    file '.command.log' into qualimap_stdout
+
+    script:
+    gcref = ''
+    if(params.genome == 'GRCh37') gcref = '-gd HUMAN'
+    if(params.genome == 'GRCm38') gcref = '-gd MOUSE'
+    """
+    qualimap bamqc $gcref \\
+    -bam $realign_bam \\
+    -outdir ${name}_qualimap \\
+    --skip-duplicated \\
+    --collect-overlap-pairs \\
+    -nt ${task.cpus} \\
+    --java-mem-size=${task.memory.toGiga()}G \\
+
+    # Print version number to standard out
+    echo "QualiMap version "\$(qualimap --version 2>&1)
+    """
+}
+
+/*
+ * Step 8 - Call Variants with HaplotypeCaller in GVCF mode (differentiate between exome and whole genome data here)
+ * 
+*/ 
+process variantCall {
+    tag "${name}"
+    publishDir "${params.outdir}/variantCalling/", mode: 'copy',
+        saveAs: {filename -> filename.replaceFirst(/variants/, "raw_variants")}
+
+    input:
+    set val(name), file(realign_bam), file(realign_bam_ind) from bam_vcall
+
+    output:
+    set val(name), file("${name}_variants.vcf"), file("${name}_variants.vcf.idx") into raw_variants
+
+    script:
+    if(params.exome){
+    """
+    gatk -T HaplotypeCaller \\
+        -I $realign_bam \\
+        -R $params.gfasta \\
+        -o ${name}_variants.vcf \\
+        -ERC GVCF \\
+        -L $params.target \\
+        --annotation HaplotypeScore \\
+        --annotation MappingQualityRankSumTest \\
+        --annotation QualByDepth \\
+        --annotation ReadPosRankSumTest \\
+        --annotation RMSMappingQuality \\
+        --annotation FisherStrand \\
+        --annotation Coverage \\
+        --standard_min_confidence_threshold_for_calling 30.0 \\
+        --dbsnp $params.dbsnp -l INFO \\
+        -variant_index_type LINEAR -variant_index_parameter 128000
+    """
+    } else { //We have a winner (genome)
+    """
+    gatk -T HaplotypeCaller \\
+        -I $realign_bam \\
+        -R $params.gfasta \\
+        -o ${name}_variants.vcf \\
+        -ERC GVCF \\
+        --annotation HaplotypeScore \\
+        --annotation MappingQualityRankSumTest \\
+        --annotation QualByDepth \\
+        --annotation ReadPosRankSumTest \\
+        --annotation RMSMappingQuality \\
+        --annotation FisherStrand \\
+        --annotation Coverage \\
+        --standard_min_confidence_threshold_for_calling 30.0 \\
+        --dbsnp $params.dbsnp -l INFO \\
+        -variant_index_type LINEAR -variant_index_parameter 128000
+    """    
+    }
+}
+
+
+/*
+ * Step 9 - Genotype generate GVCFs using GATK's GenotypeGVCFs
+ * 
+*/ 
+
+process genotypegvcfs{
+    tag "${name}"
```

Comment:
Missing `publishDir`

Reply:
Fixed in 58a6b7f72f7fa595aa867d0f1a1ccab731249996